graphql-language/src/ast/values.rs

```diff
@@ -9,45 +9,45 @@ use super::*;
 ///   - EnumValue
 ///   - ListValue[?Const]
 ///   - ObjectValue[?Const]
-pub enum Value<'a> {
-  Variable(Variable<'a>),
+pub enum Value {
+  Variable(Variable),
   Int(IntValue),
   Float(FloatValue),
   String(StringValue),
   Boolean(BooleanValue),
-  Enum(EnumValue<'a>),
-  List(ListValue<'a>),
-  Object(ObjectValue<'a>),
+  Enum(EnumValue),
+  List(ListValue),
+  Object(ObjectValue),
 }
 
 /// DefaultValue : = Value[Const]
-pub type DefaultValue<'a> = Value<'a>;
+pub type DefaultValue = Value;
 
 /// Variable : $ Name
-pub type Variable<'a> = Name<'a>;
+pub type Variable = Name;
 
 /// Name :: /[_A-Za-z][_0-9A-Za-z]*/
-pub struct Name<'a> {
+pub struct Name {
   pub loc: Option<Location>,
-  pub value: &'a str,
+  pub value: String,
```

Comment:
It would be better to remain &'a str in Name to prevent unnecessary extra allocation while parsing, even the lifetime will bubble up to ast root.